mem/idealmemcontroller/comp.go

```diff
@@ -40,11 +40,27 @@ type Comp struct {
 	*sim.TickingComponent
 
 	topPort          sim.Port
+	CtrlPort         sim.Port
 	Storage          *mem.Storage
 	Latency          int
 	addressConverter mem.AddressConverter
 
+	isDraining        bool
+	pauseIncomingReqs bool
+	currentDrainRsp   *mem.ControlMsg
```

Comment:
Do we need to keep the current responses?

Reply:
Yes, we need to keep current requests for the response.

Reply:
OK. Then, it should be `currentDrainReq`, right?

Reply:
Yes